blosc2/c2array.py

```diff
@@ -21,48 +21,83 @@
 C2SUB_URLBASE_ENVVAR = 'BLOSC_C2URLBASE'
 """Environment variable with a default Caterva2 subscriber URL base."""
 
+C2SUB_USERNAME_ENVVAR = 'BLOSC_C2USERNAME'
+"""Environment variable with a default Caterva2 subscriber user name."""
+
+C2SUB_PASSWORD_ENVVAR = 'BLOSC_C2PASSWORD'
+"""Environment variable with a default Caterva2 subscriber password."""
+
 _subscriber_data = {
-    'urlbase': None,
-    'auth_token': None,
+    'urlbase': os.environ.get(C2SUB_URLBASE_ENVVAR),
+    'auth_token': '',
 }
 """Caterva2 subscriber data saved by context manager."""
 
 
 @contextmanager
 def c2context(*, urlbase: (str | None) = None,
+              username: (str | None) = None,
+              password: (str | None) = None,
               auth_token: (str | None) = None):
     """
     Context manager that sets parameters in Caterva2 subscriber requests.
 
-    Parameters not specified or set to ``None`` simply inherit the value set
-    by the previous context manager, if any.
+    A parameter not specified or set to ``None`` inherits the value set by the
+    previous context manager, defaulting to an environment variable (see
+    below) if supported by that parameter.  Parameters set to the empty string
+    are not to be used in requests (with no default either).
+
+    If the subscriber requires authorization for requests, you may either
+    provide `auth_token` (which you should have obtained previously from the
+    subscriber), or both `username` and `password` to get that token by first
+    logging in to the subscriber.  The token will be reused until explicitly
+    reset or requested again in a latter context manager invocation.
 
     Please note that this manager is reentrant but not concurrency-safe.
 
     Parameters
     ----------
-    urlbase: str | None
+    urlbase : str | None
         A URL base that will be used when an individual C2Array instance has
-        no subscriber URL base set.  Use ``None`` to disable the base set by a
-        previous context manager (and default to the value in the
-        ``BLOSC_C2URLBASE`` environment variable, if set).
-    auth_token: str | None
+        no subscriber URL base set.  Use the ``BLOSC_C2URLBASE`` environment
```

Comment:
This is an example on why using the `C2SUB_URLBASE_ENVVAR` above is a bit pointless

Reply:
Yeah, wasn't totally convinced about how that panned out. :slightly_smiling_face: